### PR TITLE
add libraries to find broker id and zookeepers

### PR DESCRIPTION
**Pull request by TD-4242**
I created this library code for a confluent-kafka cookbook based off your default recipe that we discussed at one point of moving to a library.  You should be able to replace the ruby_block with two calls to the functions in this library.

We are testing moving away from straight Kafka so I wouldn't be able to test it enough to actually implement, but didn't want you to miss out on fixing something we planned to at some point anyway.

### CODE REVIEW

File: libraries/config_helper.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  if node['kafka']['server.properties'].key?('zookeeper.connect')
+    Chef::Log.error(
+      "broker hard set to '#{node['kafka']['server.properties']['zookeeper.connect']}' "\
+      'in server.properties node object'
+    )
+    return
+  end
+
+  if node['kafka']['zookeepers'].nil?
+    Chef::Log.error(
```
Comment (bbaugher) [inline libraries/config_helper.rb]: This should probably just be info

File: libraries/config_helper.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      broker == node['hostname']
+    end
+
+    if broker_id.nil?
+      Chef::Log.error("Unable to find #{node['fqdn']}, #{node['ipaddress']} or "\
+                      "#{node['hostname']} in node['kafka']['brokers'] : #{node['kafka']['brokers']}"
+                     )
+    else
+      node.default['kafka']['server.properties']['broker.id'] = broker_id + 1
+      Chef::Log.error("SET BROKER!! #{node['kafka']['server.properties']['broker.id']}")
```
Comment (bbaugher) [inline libraries/config_helper.rb]: Make this debug

**Comment by bbaugher** [comment]
Mind adding your name here, https://github.com/cerner/cerner_kafka/blob/master/CONTRIBUTORS.txt

**Comment by bbaugher** [comment]
+1

**Comment by mkwhitacre** [comment]
+1

**Comment by bbaugher** [comment]
Got enough +1s, merging

**Pull request merged**